be/src/storage/txn_manager.cpp

```diff
@@ -65,9 +66,17 @@ TxnManager::TxnManager(int32_t txn_map_shard_size, int32_t txn_shard_size)
     DCHECK_EQ(_txn_map_shard_size & (_txn_map_shard_size - 1), 0);
     DCHECK_EQ(_txn_shard_size & (_txn_shard_size - 1), 0);
     _txn_map_locks = new std::shared_mutex[_txn_map_shard_size];
+    ScopedCleanup txn_map_locks_release_guard([&]() { delete[] _txn_map_locks; });
```

Comment:
Can you just change those types to `std::unique_ptr<T>`?

Reply:
ok, i will use it